contributing/contribute-code.md

```diff
@@ -0,0 +1,209 @@
+     - Does the contribution conceptually fit well into Flink? Is it too much of a special case such that it makes things more complicated for the common case, or bloats the abstractions / APIs?
+     - Does the feature fit well into Flink’s architecture? Will it scale and keep Flink flexible for the future, or will the feature restrict Flink in the future?
+     - Is the feature a significant new addition (rather than an improvement to an existing part)? If yes, will the Flink community commit to maintaining this feature?
+     - Does this feature align well with Flink's roadmap and currently ongoing efforts?
+     - Does the feature produce added value for Flink users or developers? Or does it introduce the risk of regression without adding relevant user or developer benefit?
+     - Could the contribution live in another repository, e.g., Apache Bahir or another external repository?
+     - Is this a contribution just for the sake of getting a commit in an open source project (fixing typos, style changes merely for taste reasons)
+  - There is **consensus** on how to solve the problem. This includes considerations such as
+    - API and data backwards compatibility and migration strategies
+    - Testing strategies
+    - Impact on Flink's build time
+    - Dependencies and their licenses
+
+Large or controversial changes might require a [Flink Improvement Proposal (FLIP)](https://cwiki.apache.org/confluence/display/FLINK/Flink+Improvement+Proposals) or a discussion on the [dev mailing list]( {{ site.base }}/community.html#mailing-lists ) to reach agreement or consensus.
+
+Once all requirements for the ticket are met, a committer will assign somebody to the *`Assignee`* field of the ticket to work on it.
+Only committers have the permission to assign somebody.
+
+**Pull requests belonging to unassigned Jira tickets will not be reviewed or merged by the community**.
+
+
+<a name="implement"></a>
+
+### 2. Implement your change
+
+- [Set up a Flink development environment](https://cwiki.apache.org/confluence/display/FLINK/Setting+up+a+Flink+development+environment)
+- Follow the [Code Style and Quality Guide]({{ site.base }}/contributing/code-style-and-quality.html) of Flink
+- Take any discussions and requirements from the Jira issue or design document into account.
+- Do not mix unrelated issues into one contribution.
+
+
+<a name="review"></a>
+
+### 3. Open a Pull Request
+
+Code changes in Flink are reviewed and accepted through [GitHub pull requests](https://help.github.com/en/articles/creating-a-pull-request).
+
+There is a separate guide on [how to review a pull request]({{ site.base }}/contributing/reviewing-prs.html), including our pull request review process. As a code author, you should prepare your pull request to meet all requirements.
+
+Considerations before opening a pull request:
```

Comment:
Move this to the top of the section:

* Things to consider before opening the PR
* Opening the PR
* Checking that the tests pass
* How is the PR reviewed